workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -668,4 +650,73 @@ export class DynamicAuthorizationService {
     } while (paginationToken);
     return identityPermissions;
   }
+
+  private async _getAllIdentityPermissionsByIdentity(params: {
+    identityId: string;
+    identityType: IdentityType;
+  }): Promise<IdentityPermission[]> {
+    const { identityId, identityType } = params;
+    let paginationToken = undefined;
+    let identityPermissions: IdentityPermission[] = [];
+    //paginate through response
+    do {
+      // get all identity permissions associated to an identity
+      const response: GetIdentityPermissionsBySubjectResponse = await this.getIdentityPermissionsByIdentity({
+        paginationToken,
+        identityId,
+        identityType
+      });
+      paginationToken = response.paginationToken;
+      identityPermissions = identityPermissions.concat(response.data.identityPermissions);
+    } while (paginationToken);
+    return identityPermissions;
+  }
+  /**
+   * Delete all identity permissions associated to an identity
+   * @param params - requires an IdentityId, IdentityType, and AuthenticatedUser
+   *
+   * @throws - {@link RetryError} if an error is encountered during deletion
+   */
+  private async _deleteAllIdentityPermissionsByIdentity(params: {
+    identityId: string;
+    identityType: IdentityType;
+    authenticatedUser: AuthenticatedUser;
+  }): Promise<void> {
+    const { identityId, identityType, authenticatedUser } = params;
+    try {
+      const identityPermissions = await this._getAllIdentityPermissionsByIdentity({
```

Comment:
I'm not sure if we're at this level of scale yet, but there is the potential to run out of memory if you have enough identity permissions here. You don't have to change it because I don't think we're going to be getting 50k+ entries , but a paginated delete (aka request a page, fire off a delete, repeat) is usually less resource intensive and allows for a place to check if the operation should be canceled mid-execution.